docs/resources/equinix_network_device.md

```diff
@@ -102,6 +104,61 @@ resource "equinix_network_device" "panw-cluster" {
 }
 ```
 
+```hcl
+# Create self configured single Aviatrix device with cloud init file
+
+data "equinix_network_account" "sv" {
+  metro_code = "SV"
+}
+
+resource "equinix_network_device" "aviatrix-single" {
+  name            = "tf-aviatrix"
+  metro_code      = data.equinix_network_account.sv.metro_code
+  type_code       = "AVIATRIX_EDGE"
+  self_managed    = true
+  byol            = true
+  package_code    = "STD"
+  notifications   = ["john@equinix.com"]
+  term_length     = 6
+  account_number  = data.equinix_network_account.sv.number
+  version         = "6.9"
+  core_count      = 2
+  cloud_init_file = "cloudInitFileFolder/AVX-cloud-init-file.txt"
```

Comment:
Rather than taking a `cloud_init_file` parameter that contains the path to a local file, we could take a `cloud_init` parameter that has the contents of the cloud-init. Expecting that value means that a user can load the contents from a file on disk _or_ use a template to generate the cloud-init contents.  For example:
```hcl
resource "equinix_network_device" "aviatrix-with-cloudinit-from-file" {
   # ... other resource properties
   cloud_init = file("cloudInitFileFolder/AVX-cloud-init-file.txt")
}

resource "equinix_network_device" "aviatrix-with-cloudinit-from-template" {
   # ... other resource properties
   cloud_init = templatefile("cloud-init-file.tpl", {
      my_template_variable = "some_value"
   })
}
```

Reply:
Hi @ctreatma, your idea is good here. You are trying to provide a general way to allow the customer dynamically change the file content and upload it through the template. However, in this use case, we don't allow the customer to change the file contents. Those cloud init files will be downloaded directly from vendor and the customer only needs to upload the file. No changes are required and allowed here for such file uploading. I am trying to do it in the same way as we are doing for uploading the license file. 
Also, the cloud init file cannot be shared across different devices and vendors. Such template behavior is currently offered through vendor now. The customers go to the vendor site, provide some configs and then download the file directly from them. Then they come back to our platform and upload the exact file they got from vendor.

Reply:
It sounds like we're missing datasources and resources to reflect NE file management APIs:
* https://github.com/equinix/ne-go/blob/495d265a2b23c8cc255cfc6205c3cadb8785d87f/rest_file.go#L17-L20
* https://github.com/equinix/ne-go/blob/826e3e4c3868a58e09ce291ee28e991f6af1040b/rest_licensefile.go#L10-L13
(Interestingly, the "/file" endpoint seems from this code to support a 'type:licence', perhaps this endpoint alone could be used?)

With that pattern, a user might approach this as such:

```hcl
variable "filename" { default = "cloudInitFileFolder/AVX-cloud-init-file.txt" }

resource "equinix_network_device_file" "aviatrix-with-cloudinit" {
   filename = var.filename
   cloud_init = file(var.filename)
   process_type = "CLOUD_INIT"
   # metro_code
   # device_type_code
   # license_type
   # device_management_type
}

resource "equinix_network_device" "aviatrix-with-cloudinit-from-file" {
   # ... other resource properties
   cloud_init = equinix_network_device_file.aviatrix-with-cloudinit.filename
}
```

I have reservations about direct local file management within the Terraform provider, although I can't locate documentation to substantiate that.  Possible concerns are around error handling, file permissions, and the availability of the files in shared environments. I can subside those concerns with awareness of complimentary patterns including Hashicorp's own `local` file provider (https://github.com/hashicorp/terraform-provider-local/blob/main/internal/provider/data_source_local_file.go#L55-L89, https://registry.terraform.io/providers/hashicorp/local/latest/docs/resources/file) and provider config properties that reference local configuration files for keys and certificates. There may also be existing patterns in the NE TF resources that establish this practice (especially around license files).

As we consider adding yet another file reference parameter, I do wonder if we could make space for file management resources in the future, perhaps by including `_filename` in a property that will utilize the /ne/v1/files reference and omitting that suffix (or using an alternative suffix) for a property that includes local file handling.  As @ctreatma expressed, I would have expected this resource to accept direct file contents with no correlation to local filenames.  I can certainly see us deprecating the indirect file uploads in the future and asking users to adopt a pattern like the one shown above.